[docs] Add docs for new EAS dashboard security features (#31825)
expo/expo: docs/pages/accounts/audit-logs.mdx

---
title: Audit logs

> description: A list of changes that happened to selected entities owned by an account.

Reviewer: ```suggestion
description: Learn how to track and analyze your account's activities by using a set of entities.
```

Not sure if this suggestion is the accurate description but i think the goal is to surface that an account's owner or admin can view their team members' activities. Let's update this description to convey that.

---

## What are audit logs?

> Our implementation of audit logs will record actions made with Expo services by accounts. Recorded data will include what object was affected, what type of modification was made, who performed the action, and when this activity occurred.

Reviewer: ```suggestion
Audit logs record actions made with Expo Application Services (EAS) by accounts. Recorded data includes information about the affected entities, the type of modification made to them, who performed the action, and when the activity occurred.
```

Let's convey this (and the rest of the section) in a way where the feature is already present (so that the voice usage here is present when we decide to publish these docs along with the feature).

> Because audit logs can only be created and never modified or deleted, they will serve as a source of truth to help monitor events and possibly debug issues occurring within accounts and products.

Reviewer: ```suggestion
Since audit logs can only be created and never modified or deleted, they serve as a source of truth to help monitor events and debug issues occurring within accounts.
```
- Let's make this a bit more concise.
- I removed "products" since that doesn't add to any info. If we can describe it or explain it, that will be helpful.

> Audit logs are available for enterprise plan customers only. Immediately after subscribing, some of the logs will become available immediately as they could have been used internally by Expo. Some other types of logs will start to be collected after the subscription is activated.

Reviewer: ```suggestion
**Audit logs are available to Enterprise plan customers**. When subscribed, some of the logs used internally by Expo are immediately available, while other types of logs are collected after the subscription is activated.
```

- Let's emphasize the info about the Enterprise plan by making it bold
- Ideally this should be callout at the top of the page too so it is not an info missed by any user (people don't tend to read a document word by word in my experience and often skim the parts).
- Let's make it concise.

> They can be accessed for any account under [expo.dev/accounts/[account]/settings/audit-logs](https://expo.dev/accounts/[account]/settings/audit-logs).

Reviewer: ```suggestion
To access them, go to **Account settings** > [**Audit logs**](https://expo.dev/accounts/[account]/settings/audit-logs).
```

Let's add the complete path and link to the setting name. Then, this will be aligned with the path in the dashboard.

## Audit log use cases

### Permission monitoring

Audit logs can be used to track user invitations and permission changes within your organization. An example security event could include an employee account being compromised and used to invite an attacker into an organization and change their permission to admin.

> In this scenario, audit logs would record which employee account invited an account and modified permissions, and since audit logs are immutable, the attacker would not be able to delete this recorded history. Other organization members would be able to review the audit logs to determine which account was compromised and take action to revoke the attacker's permissions and secure the employee's account.

Reviewer: ```suggestion
In this scenario, audit logs would record which employee account invited the attacker and modified permissions. Since audit logs are immutable, the attacker would not be able to delete this recorded history. Other organization members will be able to review the audit logs to determine which account was compromised, take action to revoke the attacker's permissions and secure the employee's account.
```

### Access history

An Expo organization account can include many projects where development access is controlled by distribution certificates assigned to individual teams. When devices are granted to join these teams, it is important to track when access is granted and removed for historical record keeping. While a device may not currently be included in an Apple team, it may be useful to see who previously had access to the team in the event of an internal security incident.

The Apple devices listed within Expo's team settings will only show devices that are currently registered to an account, but with the creation of audit logs, historical modifications of Apple teams and devices can be viewed.

## Where are audit logs in use?

> Audit logs have been enabled for several entities that will offer the highest level of action tracking. While we will be adding more entities to audit logs in the future, at the time of this blog post, we have enabled audit logs for the following entities:

Reviewer: ```suggestion
While we are working on adding more entities in future, the following entities are already enabled:
```

- Let's make it concise since we are repeating ourselves and also remove the reference of the blog post

> - Android app credentials
> - Android keystores
> - Apple devices
> - Apple distribution certificates
> - Apple provisioning profiles
> - Apple teams
> - App store connect api keys
> - Google service account keys
> - iOS app credentials
> - User permissions
> - Apps
> - Accounts
> - User invitations

Reviewer: ```suggestion
- Accounts
- Android App Credentials
- Android Keystore
- Apple devices
- Apple Distribution Certificate
- Apple Provisioning Profile
- Apple Team
- App Store Connect API key
- Google Service Account key
- iOS App Credentials
- Project
- User Invitations
- User Permissions
```

- Let's arrange this list in alphabetical order
- I've also updated it to match the case used in the Audit logs entity type
- Not sure what "Apps" was referring to, but we do have Project (may be that is what you meant?)

## Understanding Audit Logs

Audit Log entries include the following fields:

- Actor: the account actor that performed the particular action
- Entity Type: The object that was modified
- Mutation Type: The type of modification, CREATE, UPDATE, DELETE
- Created At: When the particular action was performed

> Additionally, the three dots at the end of an Audit Log row can be opened to view any metadata relevant to the Audit Log.

Reviewer: ```suggestion
Additionally, clicking on an Audit log row, you can view the metadata relevant to that log.
```

I think our UI might have changed as I'm not seeing a three-dotted menu icon. However, clicking it to open a log's metadata seems to work.

![CleanShot 2024-10-28 at 20 12 15@2x](https://github.com/user-attachments/assets/7636eef5-b7c5-43b8-9b45-08eb3fece29c)

> ## Exporting Audit Logs

Reviewer: ```suggestion
## Exporting audit logs
```

> Audit Logs can be exported as the following file formats: CSV, JSON, JSONL. A time range must be included in the request, with a max time range of 30 days. The exported file will include all the fields shown on the Audit Logs page except for the Message field.

Reviewer: ```suggestion
Audit logs can be exported in the following file formats: CSV, JSON, and JSONL. 

A time range must be included in the request, with a maximum range of 30 days. The exported file will include all the fields shown on the Audit logs page except for the **Message** field.
```

---
title: Audit logs
description: Learn how to track and analyze your account's activities by using the audit logs.
---

import { ContentSpotlight } from '~/ui/components/ContentSpotlight';

> **info** Audit logs are available for [Enterprise plan](https://expo.dev/pricing) customers.

Audit logs record actions made with Expo Application Services (EAS) by accounts. Recorded data includes information about the affected entities, the type of modification made to them, who performed the action, and when the activity occurred.

> Since audit logs can only be created and never modified or deleted, they serve as a source of truth to help monitor events and debug issues occurring within accounts.

Reviewer: ```suggestion
## Key points

- Audit logs can only be created and never modified or deleted, they serve as a source of truth to help monitor events and debug issues occurring within accounts.
```

All the info from this paragraph until the image, is isolated factual information. Let's convert this into a list.

> **Audit logs are available to Enterprise plan customers**. When subscribed, some of the logs used internally by Expo are immediately available, while other types of logs are starting to be collected after the subscription is activated.

Reviewer: ```suggestion
- **Audit logs are available to Enterprise plan customers**. When subscribed, some of the logs used internally by Expo are immediately available, while other types of logs will be collected after the subscription is activated.
```

> Audit logs are stored for 1.5 years. In case an account is deleted, its audit logs will be deleted no later than 90 days after.

Author: Was the dash `-` intentional?

> To access them, go to **Account settings** > [**Audit logs**](https://expo.dev/accounts/[account]/settings/audit-logs).

Reviewer: ```suggestion
- To access them, go to **Account settings** > [**Audit logs**](https://expo.dev/accounts/[account]/settings/audit-logs).
```
